### PR TITLE
Pull request and code merg

### PR DESCRIPTION
**Pull request by itisusama**
This is first pull request

### CODE REVIEW

File: app/Http/Controllers/ProductController.php [hunk 1]
```diff
@@ -11,5 +11,12 @@ class ProductController extends Controller
     public function listProduct($id=null){
         return $id?Product::find($id):Product::all();
     }
+
+    //Searching for Categories
+    public function searchCategory($id){
```
Comment (UmerShahid1012) [inline app/Http/Controllers/ProductController.php:16]: why are you using DB queries?
why you are not using Eloquent query?

File: routes/api.php [hunk 2]
```diff
@@ -10,4 +10,5 @@
 });
 
 Route::get('listCategory',[CategoryController::class,'listCategory']);
-Route::get('listProduct/{id?}',[ProductController::class,'listProduct']);
\ No newline at end of file
+Route::get('listProduct/{id?}',[ProductController::class,'listProduct']);
```
Comment (UmerShahid1012) [inline routes/api.php:13]: where is route names?

File: app/Http/Controllers/ProductController.php [hunk 3]
```diff
@@ -11,5 +11,12 @@ class ProductController extends Controller
     public function listProduct($id=null){
```
Comment (UmerShahid1012) [inline app/Http/Controllers/ProductController.php:11]: where are function comments ?

File: app/Http/Controllers/ProductController.php (hunk 1 above)
Comment (itisusama) [reply app/Http/Controllers/ProductController.php:16]: Sir I did the change in second baranch

File: routes/api.php (hunk 2 above)
Comment (itisusama) [reply routes/api.php:13]: I gave route names in second branch

File: app/Http/Controllers/ProductController.php (hunk 3 above)
Comment (itisusama) [reply app/Http/Controllers/ProductController.php:11]: I commented everything in second branch

**Pull request merged**